kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java

```diff
@@ -76,24 +79,32 @@ public FileSystemViewKeyInputFormat(FileSystemDataset<E> dataset,
     Format format = dataset.getDescriptor().getFormat();
 
     boolean isSpecific = SpecificRecord.class.isAssignableFrom(dataset.getType());
+    String readerSchema = conf.get(KITE_READER_SCHEMA);
+
+    Preconditions.checkState(!isSpecific || readerSchema == null,
+      "Illegal configuration: {} must not be set if dataset uses a specific type {}",
+      KITE_READER_SCHEMA, dataset.getType());
```

Comment:
If the reader schema property is private and we don't allow the ConfigBuilder to set both, then I'm fine with using a warning here instead of failing the job, with the specific object taking precedence.